howto/getting-started/deploy-workload-manager.md

```diff
@@ -0,0 +1,163 @@
+
+### Create a model for the cluster
+
+To create a Juju model for your cluster, use the following command:
+
+```shell
+juju add-model charmed-hpc
+```
+
+Now you're ready to deploy the workload manager for your Charmed HPC cluster!
+
+## Deploy Slurm
+
+Slurm can be deployed multiple ways using Juju.
+
+```````{tabs}
+
+``````{group-tab} CLI
+
+To deploy Slurm via the Juju CLI, use the following commands:
+
+```shell
+# Deploy Slurm services.
+juju deploy slurmctld --constraints="virt-type=virtual-machine"
+juju deploy slurmd --constraints="virt-type=virtual-machine"
+juju deploy slurmdbd --constraints="virt-type=virtual-machine"
+juju deploy slurmrestd --constraints="virt-type=virtual-machine"
+juju deploy mysql --channel "8.0/stable"
+juju deploy mysql-router slurmdbd-mysql-router --channel "dpe/beta"
+
+# Integrate services together.
+juju integrate slurmctld:slurmd slurmd:slurmctld
+juju integrate slurmctld:slurmdbd slurmdbd:slurmctld
+juju integrate slurmctld:slurmrestd slurmrestd:slurmctld
+juju integrate slurmdbd-mysql-router:backend-database mysql:database
+juju integrate slurmdbd:database slurmdbd-mysql-router:database
+```
+
+``````
+
+``````{group-tab} Bundle
+
+`````{tabs}
+
+````{group-tab} Charmhub
+
+To deploy Slurm via the Slurm bundle published on [Charmhub](https://charmhub.io/slurm),
+use the following command:
+
+```shell
+juju deploy slurm
+```
+
+````
+
+````{group-tab} bundle.yaml
+
+To deploy Slurm via a Juju bundle file, first open a file named _bundle.yaml_ in
+your favourite text editor, and enter the following YAML content:
+
+```yaml
+description: Deploy a ready-to-go Slurm cluster.
+series: jammy
+name: slurm
+applications:
+  slurmctld:
+    charm: slurmctld
+    constraints: virt-type=virtual-machine
+  slurmd:
+    charm: slurmd
+    constraints: virt-type=virtual-machine
+  slurmdbd:
+    charm: slurmdbd
+    constraints: virt-type=virtual-machine
+  slurmrestd:
+    charm: slurmrestd
+    constraints: virt-type=virtual-machine
+  mysql:
+    charm: mysql
+    channel: 8.0/stable
+  slurmdbd-mysql-router:
+    charm: mysql-router
+    channel: dpe/beta
+relations:
+- - slurmctld:slurmd
+  - slurmd:slurmctld
+- - slurmctld:slurmdbd
+  - slurmdbd:slurmctld
+- - slurmctld:slurmrestd
+  - slurmrestd:slurmctld
+- - slurmdbd-mysql-router:backend-database
+  - mysql:database
+- - slurmdbd:database
+  - slurmdbd-mysql-router:database
+```
+
+Save and close the file after entering the YAML content, and use the following
+command to deploy Slurm using your custom _bundle.yaml_ file:
+
+```shell
+juju deploy ./bundle.yaml
+```
+
+````
+
+`````
+
+``````
+
+```````
+
+```{note}
+The instructions above pass `virt-type=virtual-machine` as a constraint to the Slurm charms
+to instruct LXD to provide a virtual machine rather than a system container. Slurm does not
+fully work within system containers unless some configuration modifications are applied to
+the default LXD profile.
```

Comment:
This should probably be removed before the deployment steps, because I think it's important to really highlight that the machines only work correctly using virtual-machine.

Reply:
You know, I originally had note before the deployment steps but then moved it down below the the deployment steps. In the future we should potentially include a link out to the explanation documentation that goes over why we don't support system containers and instead direct LXD to use virtual machines.